sasmodels/models/micromagnetic_FF_3D.py

```diff
@@ -0,0 +1,200 @@
+            B(H): weighting function for purely magnetic scattering I_mag(q,H)  due to misaligned magnetic moments, different for the various possible spin-resolved scattering cross sections
+            C: weighting function for nuclear-magnetic interference scattering
+            F_N: nuclear form factor
+            F_M: magnetic form factor
+The underlying defect can have a core-shell structure.
+"""
+category = "shape:sphere"
+
+# pylint: disable=bad-whitespace, line-too-long
+#             ["name", "units", default, [lower, upper], "type","description"],
+parameters = [["nuc_radius", "Ang", 50., [0, inf], "volume", "Structural radius of the core"],
+              ["nuc_thickness", "Ang", 40., [0, inf], "volume", "Structural thickness of shell"],  
+              ["mag_radius", "Ang", 50., [0, inf], "volume", "Magnetic radius of the core"],
+              ["mag_thickness", "Ang", 40., [0, inf], "volume", "Magnetic thickness of shell"], 
+              ["hk_radius", "Ang", 50., [0, inf], "volume", "Anisotropy radius of the core"],            
+              ["nuc_sld_core", "1e-6/Ang^2", 1.0, [-inf, inf], "", "Core scattering length density"],
+              ["nuc_sld_shell", "1e-6/Ang^2", 1.7, [-inf, inf], "", "Scattering length density of shell"],
+              ["nuc_sld_solvent", "1e-6/Ang^2", 6.4, [-inf, inf], "", "Solvent scattering length density"],
+              ["mag_sld_core",    "1e-6/Ang^2", 1.0,  [-inf, inf], "",    "Magnetic scattering length density of core"],
+              ["mag_sld_shell", "1e-6/Ang^2", 1.7, [-inf, inf], "", "Magnetic scattering length density of shell"],
+              ["mag_sld_solvent", "1e-6/Ang^2", 3.0,  [-inf, inf], "",  "Magnetic scattering length density of solvent"],
+              ["hk_sld_core", "1e-6/Ang^2", 1.0,  [0, inf], "",    "Anisotropy field of defect"],
+              ["Hi", "T", 2.0,  [0, inf], "",    "Effective field inside the material"],
+              ["Ms", "T", 1.0,  [0, inf], "",    "Volume averaged saturation magnetisation"], 
+              ["A", "pJ/m", 10.0,  [0, inf], "",    "Average exchange stiffness constant"],  
+              ["D", "mJ/m^2", 0.0,  [0, inf], "",    "Average DMI constant"],                          
+              ["up_i",          "None",  0.5, [0, 1], "", "Polarisation incoming beam"],
+              ["up_f",          "None",  0.5, [0, 1], "", "Polarisation outgoing beam"],
+              ["alpha",          "None",  90, [0, 180], "", "Inclination of field to neutron beam"],
+              ["beta",          "None",  0, [0, 360], "", "Rotation of field around neutron beam"],             
+             ]
+# pylint: enable=bad-whitespace, line-too-long
+
+
+
+
+source = ["lib/sas_3j1x_x.c", "lib/core_shell.c", "lib/gauss76.c", "lib/magnetic_functions.c", "micromagnetic_FF_3D.c"]
+structure_factor = False
+have_Fq = False
+single=False
```

Comment:
This is already the default for these values, so there is no need to include them.

Reply:
`single` defaults to True.

Reply:
use `opencl = False` here rather than suppressing OpenCL in `.github/workflows/test.yml`

Reply:
Done